### PR TITLE
Update liblouis to 3.31

### PR DESCRIPTION
**Pull request by LeonarddeR**
## Preliminary note

Liblouis 3.31 is to be released on the 2nd of September

### Link to issue number:
None

### Summary of the issue:

Liblouis 3.31 is to be released soon. AMong new tables, this release contains a bug fix for Spanish causing digits to be presented incorrectly, see https://github.com/liblouis/liblouis/pull/1624. CC @jmdaweb. Hence this pr is filed against beta.In my humble opinion, we are still early enough in the beta cycle to warrant this update.

### Description of user facing changes
Update to Liblouis 3.31, adding new tables as mentioned in the changelog.

### Description of development approach
Update the submodule, added and updated definitions in `brailleTables.py`

### Testing strategy:
Running NVDA< ensured that the new and changed tables work.

### Known issues with pull request:
None known

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Upgraded LibLouis Braille translator to version 3.31.0, adding new Braille tables for Thai grade 1 and Greek international Braille with single-cell accented letters.
	- Updated eSpeak NG component to version 1.52-dev, now supporting Faroese and Xextan languages.
	- Enhanced command line options for better user experience during updates.

- **Bug Fixes**
	- Resolved a bug related to back-translation of numbers in Spanish Braille.

- **Documentation**
	- Updated documentation to reflect changes in Braille table names and versions.

### CODE REVIEW

**Comment by LeonarddeR** [comment]
Cc @codeofdusk who did the liblouis updates in the past.

**Comment by codeofdusk** [comment]
Liblouis 3.31 has been [officially released](https://github.com/liblouis/liblouis/releases/tag/v3.31.0). I also vote for its inclusion into NVDA 2024.4.

**Comment by codeofdusk** [comment]
CC @seanbudd.

**Comment by seanbudd** [comment]
@codeofdusk @LeonarddeR - we don't introduce risky changes, new features or dependency updates in the beta cycle unless they are important bug fixes.
Please see our [release process docs](https://github.com/nvaccess/nvda/blob/master/projectDocs/community/releaseProcess.md#beta-phase) - the beta cycle is for:

- addressing regressions introduced in this release.
- addressing a bug in a "must have" feature for this release.
- addressing a critical Operating System (or software) change out of our control.

Generally we would expect these updates to happen early in the alpha cycle to ensure there is maximum testing before we release.

**Comment by codeofdusk** [comment]
This PR *does* include an important bug fix:

> this release contains a bug fix for Spanish causing digits to be presented incorrectly, see liblouis/liblouis#1624. CC @jmdaweb. Hence this pr is filed against beta.In my humble opinion, we are still early enough in the beta cycle to warrant this update.

**Comment by seanbudd** [comment]
Is this a bug fix for a recent regression? Most liblouis releases contain some level of bug fixes, we would still generally hold back from a late inclusion as regressions may be introduced with these bug fixes.

If we merge this late, and there's regressions within the bug fixes, we get caught in a bad cycle where we have to revert the update and extend the release cycle. Or even worse, we ship with regressions. If this is merged early in a release cycle, it can get adequate testing, and we can ship it with greater certainty that is doesn't introduce any regressions.

**Comment by codeofdusk** [comment]
Per https://github.com/liblouis/liblouis/pull/1574#issuecomment-2305408543 this is new as of Liblouis 3.30 (NVDA 2024.3).

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -40,8 +42,16 @@ There have also been a number of fixes, including to mouse tracking in Firefox,
 ### Changes
 
 * The `-c`/`--config-path` and `--disable-addons` command line options are now respected when launching an update from within NVDA. (#16937)
-* eSpeak NG has been updated to 1.52-dev commit `961454ff`. (#16775)
-  * Added new languages Faroese and Xextan.
+* Component updates:
+  * Updated LibLouis braille translator to [3.31.0](https://github.com/liblouis/liblouis/releases/tag/v3.31.0). (#17080, @LeonarddeR)
```
Comment (seanbudd) [inline user_docs/en/changes.md:46]: can the bug fix be mentioned here?

**Comment by codeofdusk** [comment]
Given @LeonarddeR stepping away (and inability to respond to mentions) I'm happy to take this over.

**Comment by codeofdusk** [comment]
@seanbudd This PR is now ready for review against beta, but since I don't own it, I can't change its status.

**Comment by LeonarddeR** [comment]
> Given @LeonarddeR stepping away (and inability to respond to mentions) I'm happy to take this over.

Thanks @codeofdusk

File: user_docs/en/changes.md [hunk 2]
```diff
@@ -40,8 +42,17 @@ There have also been a number of fixes, including to mouse tracking in Firefox,
 ### Changes
 
 * The `-c`/`--config-path` and `--disable-addons` command line options are now respected when launching an update from within NVDA. (#16937)
-* eSpeak NG has been updated to 1.52-dev commit `961454ff`. (#16775)
-  * Added new languages Faroese and Xextan.
+* Component updates:
+  * Updated LibLouis Braille translator to [3.31.0](https://github.com/liblouis/liblouis/releases/tag/v3.31.0). (#17080, @LeonarddeR, @codeofdusk)
+    * Fixed back-translation of numbers in Spanish Braille.
```
Comment (LeonarddeR) [inline user_docs/en/changes.md:47]: I don't think the fix is only for back-translation, I"m not sure though.
```suggestion
    * Fixed translation of numbers in Spanish Braille.
```

**Comment by codeofdusk** [comment]
From `news`:

> A change to the Spanish back-translation that went into the last release was reverted as it caused more problems than it solved.

**Comment by LeonarddeR** [comment]
If it is only back translation, then I guess it is of slightly less importance. Honestly I thought that it caused issues with forward translation.

**Comment by jmdaweb** [comment]
It causes issues with forward translation as well. The prefix for digits (`#` symbol) has disappeared on the spanish grade 1 table. This pr is quite urgent and should be included in the next beta. We have a lot of complains and users expect this bug to be fixed in 2024.4.

**Review by SaschaCowley — APPROVED** [review]

**Comment by codeofdusk** [comment]
@SaschaCowley Please re-target this PR to the `beta` branch. I don't own this PR, so can't change its base branch.

**Pull request merged**